### PR TITLE
rex: Fix compilation on Linux

### PR DESCRIPTION
**Pull request by iMichka**
and add LWP::UserAgent for Linux too

- [x] Have you followed the [guidelines for contributing](https://github.com/Homebrew/homebrew-core/blob/HEAD/CONTRIBUTING.md)?
- [x] Have you checked that there aren't other open [pull requests](https://github.com/Homebrew/homebrew-core/pulls) for the same formula update/change?
- [ ] Have you built your formula locally with `brew install --build-from-source <formula>`, where `<formula>` is the name of the formula you're submitting?
- [ ] Is your test running fine `brew test <formula>`, where `<formula>` is the name of the formula you're submitting?
- [ ] Does your build pass `brew audit --strict <formula>` (after doing `brew install <formula>`)?

-----

### CODE REVIEW

**Comment by iMichka** [comment]
I think perl has different versions now. How can I handle the `"INC=-I#{MacOS.sdk_path}/System/Library/Perl/5.18/darwin-thread-multi-2level/CORE"` path on macOS?

File: Formula/rex.rb [hunk 1]
```diff
@@ -234,8 +232,13 @@ def perl_build
       system "./Build", "PERL5LIB=#{ENV["PERL5LIB"]}"
       system "./Build", "install"
     elsif File.exist? "Makefile.PL"
-      system "perl", "Makefile.PL", "INSTALL_BASE=#{libexec}",
+      on_macos do
+        system "perl", "Makefile.PL", "INSTALL_BASE=#{libexec}",
                      "INC=-I#{MacOS.sdk_path}/System/Library/Perl/5.18/darwin-thread-multi-2level/CORE"
```
Comment (carlocab) [inline Formula/rex.rb:237]:
```suggestion
                     "INC=-I#{MacOS.sdk_path}/System/Library/Perl/#{MacOS.preferred_perl_version}/darwin-thread-multi-2level/CORE"
```

**Comment by mitchblank** [comment]
I wouldn't let the OS/X failures block this; that is a known issue due to TermReadKey problems.  I had 50% of a fix working at one point but got distracted before I finished a PR :-(

**Comment by iMichka** [comment]
Ok I'll merge this then. My focus is aligning the linux and Mac versions. If the build issue is known from before, I will not invest more time in fixing it.

**Pull request merged**